pkg/yamlmeta/node.go

```diff
@@ -57,8 +58,8 @@ func (m *Map) SetValue(val interface{}) error {
 }
 
 func (mi *MapItem) SetValue(val interface{}) error {
```

Comment:
what about Document's SetValue?

Reply:
Yeah, I thought about it for a minute...

This whole thing came about because within a template, one can break YAML structure through an expression. But AFAIK, there's no way to make the value of a Document contain a Document or DocumentSet.

But without it, there's this question of whether that omission is design intent, partial implementation, or a miss.

I think it's worth clearing up that ambiguity. 👍🏼

Reply:
> But AFAIK, there's no way to make the value of a Document contain a Document or DocumentSet.

```yaml
--- #@ docset()
```

Reply:
> ``` 
> --- #@ docset()
> ```

As I was writing that sentence I thought to myself, "and saying this is likely the fastest way to learn how to do so." 😬 

And yes, of course, we do this constantly when we inject the evaluated contents of a private library... :)